crates/rust-analyzer/src/handlers.rs

```diff
@@ -1062,7 +1062,7 @@ pub(crate) fn handle_call_hierarchy_outgoing(
     let item = params.item;
 
     let doc = TextDocumentIdentifier::new(item.uri);
-    let frange = from_proto::file_range(&snap, doc, item.range)?;
+    let frange = from_proto::file_range(&snap, doc, item.selection_range)?;
```

Comment:
I wonder if that means that our API and/or protocol are off. Positions only make sense for the first request which initializes call hierarchy. Navingating to incoming/outgoing call should not use positions. At the same time, round-tripping some kind of persistent cookies seems awkward. 

I guess, for the time being this is a fine fix, but this needs more design long-term.

Reply:
I was thinking of reconstructing the call item and passing that down and may try something like that in a followup PR.